apps/api/src/app/shared/services/in-memory-provider/in-memory-provider.service.ts

```diff
@@ -0,0 +1,73 @@
+import { Logger } from '@nestjs/common';
+
+import { getRedisInstance, getRedisProviderConfig, IRedisProviderConfig, Redis } from './redis-provider';
+
+import { ApiException } from '../../exceptions/api.exception';
+
+const LOG_CONTEXT = 'InMemoryCluster';
+
+export type InMemoryProviderClient = Redis;
+type InMemoryProviderConfig = IRedisProviderConfig;
+
+export class InMemoryProviderService {
+  public inMemoryProviderClient: InMemoryProviderClient;
+  public inMemoryProviderConfig: InMemoryProviderConfig;
+
+  constructor() {
+    if (!this.inMemoryProviderClient) {
+      this.inMemoryProviderClient = process.env.IN_MEMORY_CLUSTER_MODE
+        ? this.inMemoryClusterProviderSetup()
+        : this.inMemoryProviderSetup();
+    }
```

Comment:
From a very quick look at this, it seems as if Caching is required to run novu, where previously we had an optional for this (You can run novu without the cache service, if you haven't passed the REDIS_HOST params).

How the current behaviour will work? Can we still allow running novu without a cache service?